### PR TITLE
DOC: Add Metrics Tables To Documentation

### PR DESCRIPTION
**Pull request by simongraham**
Added tables of metrics to the documentation of TIAToolbox.

- [x] HoVer-Net results
- [x] IDaRS results
- [x] Patch predictor results
- [x] Semantic segmentation results

### CODE REVIEW

File: tiatoolbox/models/engine/patch_predictor.py [hunk 1]
```diff
@@ -44,6 +44,90 @@ def __init__(
...
     """Patch-level predictor.
 
+    The models provided by tiatoolbox should give the following results:
+
+    .. list-table:: PatchPredictor performance on the Kather100K dataset [1]
+       :widths: 15 15
+       :header-rows: 1
+
+       * - Model name
+         - F$_1$ score
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/patch_predictor.py:54]:
```suggestion
         - F_1 score
```

File: tiatoolbox/models/engine/patch_predictor.py [hunk 2]
```diff
@@ -44,6 +44,90 @@ def __init__(
...
+         - 0.992
+       * - googlenet-kather100k
+         - 0.992
+    
+    .. list-table:: PatchPredictor performance on the PCam dataset [2]
+       :widths: 15 15
+       :header-rows: 1
+
+       * - Model name
+         - F$_1$ score
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/patch_predictor.py:95]:
```suggestion
         - F_1 score
```

File: tiatoolbox/models/engine/patch_predictor.py (hunk 1 above)
Comment (shaneahmed) [reply tiatoolbox/models/engine/patch_predictor.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/335/commits/ad144172ff82bc40d3dceca5fea8c7adadd7eb48

File: tiatoolbox/models/engine/patch_predictor.py (hunk 2 above)
Comment (shaneahmed) [reply tiatoolbox/models/engine/patch_predictor.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/335/commits/ad144172ff82bc40d3dceca5fea8c7adadd7eb48

File: tiatoolbox/models/engine/patch_predictor.py [hunk 3]
```diff
@@ -44,6 +44,90 @@ def __init__(
...
     """Patch-level predictor.
 
+    The models provided by tiatoolbox should give the following results:
+
+    .. list-table:: PatchPredictor performance on the Kather100K dataset [1]
+       :widths: 15 15
+       :header-rows: 1
+
+       * - Model name
+         - F_1 score
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/patch_predictor.py:54]: Can we add this as F1 or with "1" as a subscript?

File: tiatoolbox/models/engine/patch_predictor.py [hunk 4]
```diff
@@ -44,6 +44,90 @@ def __init__(
...
+         - 0.992
+       * - googlenet-kather100k
+         - 0.992
+
+    .. list-table:: PatchPredictor performance on the PCam dataset [2]
+       :widths: 15 15
+       :header-rows: 1
+
+       * - Model name
+         - F_1 score
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/patch_predictor.py:95]: Can we add this as F1 or with "1" as a subscript?

File: tiatoolbox/models/engine/semantic_segmentor.py [hunk 5]
```diff
@@ -351,6 +351,35 @@ def __getitem__(self, idx: int):
 class SemanticSegmentor:
     """Pixel-wise segmentation predictor.
 
+    The tiatoolbox model should produce the following results on the BCSS dataset
+    using fcn_resnet50_unet-bcss.
+
+    .. list-table:: Semantic segmentation performance on the BCSS dataset
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/semantic_segmentor.py:357]:
```suggestion
    .. list-table:: Semantic segmentation performance on the BCSS dataset.
```

File: tiatoolbox/models/engine/patch_predictor.py (hunk 3 above)
Comment (John-P) [reply tiatoolbox/models/engine/patch_predictor.py:54]: https://docutils.sourceforge.io/docs/ref/rst/roles.html#subscript

**Review by John-P — APPROVED** [review]
I am happy with this after addressing the comments from Shan.

File: tiatoolbox/models/engine/semantic_segmentor.py (hunk 5 above)
Comment (simongraham) [reply tiatoolbox/models/engine/semantic_segmentor.py:357]: Sentence follows on next line, so we shouldn't be using a full stop.

File: tiatoolbox/models/engine/patch_predictor.py [hunk 6]
```diff
@@ -44,6 +44,90 @@ def __init__(
...
     """Patch-level predictor.
 
+    The models provided by tiatoolbox should give the following results:
+
+    .. list-table:: PatchPredictor performance on the Kather100K dataset [1]
+       :widths: 15 15
+       :header-rows: 1
+
+       * - Model name
+         - F:sub:`1` score
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/patch_predictor.py:54]:
```suggestion
         - F\ :sub:`1`\ score
```

File: tiatoolbox/models/engine/patch_predictor.py (hunk 3 above)
Comment (shaneahmed) [reply tiatoolbox/models/engine/patch_predictor.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/335/commits/018a504fc133ef9bfa5c2753e0fb3aacfb681cb9

File: tiatoolbox/models/architecture/idars.py [hunk 7]
```diff
@@ -14,14 +14,48 @@
 
 
 class IDaRS(CNNModel):
-    """Retrieve the model and add custom preprocessing used in IDaRS paper.
+    """Initialise IDaRS and add custom preprocessing as used in the original paper [1].
+
+    The tiatoolbox model should produce the following results:
+
+    .. list-table:: IDaRS performance
```
[the remark this answers is not in the document]
Comment (shaneahmed) [reply tiatoolbox/models/architecture/idars.py]: Resolved in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/335/commits/68871ee98596da7ddb90c1b5f82c4ee1454bdde1

**Pull request merged**